### PR TITLE
feat(lld): Impletation of variant B for analytics FF

### PR DESCRIPTION
**Pull request by LucasWerey**
### 📝 Description

Implementation of the variant B of the analytics feature flag 
To see it you have to reset your LLD (so  hasSeenAnalyticsOptInPrompt will be false) and activate the feature flag lldAnalyticsOptInPrompt in the settings with the following override : 
`{"enabled":true,"params":{"variant":"B","entryPoints":["Onboarding","Portfolio"]}}`

Demo welcome page : 

https://github.com/LedgerHQ/ledger-live/assets/73439207/f9c1fe5c-83bc-462d-911d-00be897f9bb2

Demo portfolio : 

https://github.com/LedgerHQ/ledger-live/assets/73439207/e768e43c-4fbd-4c55-be73-97ddb1a5450f

### ❓ Context

- **JIRA or GitHub link**:[LIVE-11680](https://ledgerhq.atlassian.net/browse/LIVE-11680) and [LIVE-11533](https://ledgerhq.atlassian.net/browse/LIVE-11533) 

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-11680]: https://ledgerhq.atlassian.net/browse/LIVE-11680?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-11533]: https://ledgerhq.atlassian.net/browse/LIVE-11533?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/hooks/useVariantB.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    dispatch(setShareAnalytics(true));
+    goToPersonalizedRecommandations();
+    trackAnalyticsClick(true);
+  };
+
+  const clickOnRefuseAnalytics = () => {
+    dispatch(setShareAnalytics(false));
+    goToPersonalizedRecommandations();
+    trackAnalyticsClick(false);
+  };
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/hooks/useVariantB.tsx:66]:
```suggestion
  const setupAnalytics = (value: boolean) => {
    dispatch(setShareAnalytics(value));
    goToPersonalizedRecommandations();
    trackAnalyticsClick(value);
  };
  
  const acceptAnalytics = () => setupAnalytics(true)
  const rejectAnalytics = () => setupAnalytics(false)
 
```

File: apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/hooks/useVariantB.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    dispatch(setSharePersonalizedRecommandations(true));
+    onSubmit?.();
+    trackPersonalizedExperienceClick(true);
+  };
+
+  const clickOnRefusePersonalizedExperience = () => {
+    dispatch(setSharePersonalizedRecommandations(false));
+    onSubmit?.();
+    trackPersonalizedExperienceClick(false);
+  };
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/hooks/useVariantB.tsx:91]: Same as above :)

File: apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/screens/VariantB/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  return (
+    <>
+      <Flex flexDirection={"column"} mx={"40px"} height={"100%"} pt={"40"}>
+        {(() => {
+          if (currentStep === 0) {
+            return <AnalyticsScreen currentTheme={currentTheme} />;
+          } else if (currentStep === 1) {
+            return <RecommandationsScreen currentTheme={currentTheme} />;
+          }
+        })()}
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/screens/VariantB/index.tsx:37]: As you have only 2 steps : 

```suggestion
        {currentStep === 0 ? 
        <AnalyticsScreen currentTheme={currentTheme} />  : <RecommandationsScreen currentTheme={currentTheme} />
          }

```

**Review by mcayuelas-ledger — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/screens/VariantB/components/Footer/index.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+interface VariantBFooterProps {
+  currentStep: number;
+  clickOptions: { [key: string]: () => void };
+}
+
+const VariantBFooter = ({ currentStep, clickOptions }: VariantBFooterProps) => {
+  const { t } = useTranslation();
+
+  const handleAcceptClick = () => {
```
Comment (cgrellard-ledger) [inline apps/ledger-live-desktop/src/newArch/AnalyticsOptInPrompt/screens/VariantB/components/Footer/index.tsx:14]: Can we move this logic in the hook ?

**Review by cgrellard-ledger — APPROVED** [review]

**Pull request merged**